### ISSUE TITLE
Issue: Assertion `cur_target >= 0 && cur_target < n_classes' failed.  at /tmp/luarocks_nn-scm-1-6864/nn/lib/THNN/generic/ClassNLLCriterion.c:62

### ISSUE THREAD

**Issue by nandofernandesneto** [torch/torch7#1050]
While running the training code for my CNN, I'm getting the following error, after some batches:
fernando@fernando-VirtualBox:~/captcha2$ th captcha.lua
parameters size ..	
 18476404
[torch.LongStorage of size 1]

Iteration: 1 | Batch: 1 | Loss: 3.5977624149744	
Iteration: 1 | Batch: 2 | Loss: 3.6014592890256	
Iteration: 1 | Batch: 3 | Loss: 3.6013539505785	
Iteration: 1 | Batch: 4 | Loss: 3.5953609041295	
Iteration: 1 | Batch: 5 | Loss: 3.6038447292168	
Iteration: 1 | Batch: 6 | Loss: 3.6025368439634	
Iteration: 1 | Batch: 7 | Loss: 3.5915846844744	
Iteration: 1 | Batch: 8 | Loss: 3.5976407858355	
Iteration: 1 | Batch: 9 | Loss: 3.5735840286763	
Iteration: 1 | Batch: 10 | Loss: 3.5527650301029	
Iteration: 1 | Batch: 11 | Loss: 3.5983930120699	
Iteration: 1 | Batch: 12 | Loss: 3.5834157804777	
Iteration: 1 | Batch: 13 | Loss: 3.5943281107252	
Iteration: 1 | Batch: 14 | Loss: 3.5890408629174	
Iteration: 1 | Batch: 15 | Loss: 3.5682873484946	
Iteration: 1 | Batch: 16 | Loss: 3.5957385018047	
Iteration: 1 | Batch: 17 | Loss: 3.5735059927977	
Iteration: 1 | Batch: 18 | Loss: 3.6068114363917	
Iteration: 1 | Batch: 19 | Loss: 3.5887971715796	
Iteration: 1 | Batch: 20 | Loss: 3.5758407942197	
/home/fernando/torch/install/bin/luajit: /home/fernando/torch/install/share/lua/5.1/nn/THNN.lua:110: Assertion `cur_target >= 0 && cur_target < n_classes' failed.  at /tmp/luarocks_nn-scm-1-6864/nn/lib/THNN/generic/ClassNLLCriterion.c:62
stack traceback:
	[C]: in function 'v'
	/home/fernando/torch/install/share/lua/5.1/nn/THNN.lua:110: in function 'ClassNLLCriterion_updateOutput'
	...ndo/torch/install/share/lua/5.1/nn/ClassNLLCriterion.lua:41: in function 'updateOutput'
	...torch/install/share/lua/5.1/nn/CrossEntropyCriterion.lua:20: in function 'updateOutput'
-----------------------------------------------------------------------------------------------
It is also worth mentioning that this issue just appeared when I've changed my training set. Is this a bug?

Moreover, the number of batches that it runs also fluctuates - I imagine due to the fact that I'm training the CNN with SGD, which is stochastic by nature.

Can anyone help me?

Thanks in advance,

Fernando Fernandes

**Comment by clairett** [comment]
I also came across this error, it seems like your test data has label out of training data's label range

**Comment by abhanjac** [comment]
I made a network to classify two classes, and it was working fine. Then I added 3 more classes and it started throwing this error. I found out that I forgot to change the number of output of the final fully connected layer from 2 to 5. Once I changed that everything started working.
